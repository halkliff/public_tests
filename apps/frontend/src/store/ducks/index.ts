import { combineReducers } from 'redux';
import client, { StateData as ClientState } from './client';

export interface GlobalState {
  client: ClientState;
}

export default combineReducers<GlobalState>({ client });
