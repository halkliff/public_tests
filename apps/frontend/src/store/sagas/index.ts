import { takeEvery } from 'redux-saga/effects';

export default function*() {
  yield takeEvery({ take: () => {} }, () => {});
}
