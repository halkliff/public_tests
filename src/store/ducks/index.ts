import { combineReducers } from "redux";
import phoneData from "./phoneData";

export default combineReducers({
  phoneData
});
