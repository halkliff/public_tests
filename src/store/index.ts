import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './ducks';
import rootSagas from './sagas';

const sagaMiddleware = createSagaMiddleware();

const middleWares = [sagaMiddleware];

const store = createStore(rootReducer, applyMiddleware(...middleWares));

sagaMiddleware.run(rootSagas);

export default store;
