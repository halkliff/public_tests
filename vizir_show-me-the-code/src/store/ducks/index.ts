import { combineReducers } from 'redux';
import phoneData, { StateData as PhoneStateData } from './phoneData';

export interface GlobalState {
  phoneData: PhoneStateData;
}

export default combineReducers<GlobalState>({
  phoneData
});
