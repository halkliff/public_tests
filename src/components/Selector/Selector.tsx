import React, { FunctionComponent, useState } from 'react';
import { PhoneTransactionType } from 'store/ducks/phoneData';
import {
  Select,
  FormControl,
  MenuItem,
  Button,
  OutlinedInput,
  InputLabel
} from '@material-ui/core';
import { Search } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector } from 'react-redux';
import { GlobalState } from 'store/ducks';
import { appPalette } from 'styles/jss';

const selectorStyles = makeStyles({
  section: {
    padding: '1rem 1.5rem',
    flex: 1,
    justifyContent: 'center',
    display: 'grid',
    gridTemplateColumns: '4fr 2fr 2fr 1fr 4fr',
    gridColumnGap: '1.5rem',
    '& input:focused fieldset': {
      borderColor: appPalette.secondaryColor.main
    }
  },
  selector1: {
    gridColumn: '2/3'
  },
  selector2: {
    gridColumn: '3/4'
  },
  button: {
    gridColumn: '4/5',
    margin: '.5rem 0'
  }
});

/**
 * Function to map the origins from a given destiny
 * @param transactions The existing transactions in the state
 * @param destiny The destiny we're looking for.
 */
export function mapOriginsFromDestiny(
  transactions: PhoneTransactionType[],
  destiny: string
) {
  const mappedOrigins: string[] = [];

  transactions.forEach(value => {
    if (value.phoneData.destino === destiny)
      mappedOrigins.push(value.phoneData.origem);
  });

  return mappedOrigins;
}

/**
 * Function to map the destinies from a given destiny
 * @param transactions The existing transactions in the state
 * @param origin The origin we're looking for.
 */
export function mapDestiniesFromOrigin(
  transactions: PhoneTransactionType[],
  origin: string
) {
  const mappedDestinies: string[] = [];

  transactions.forEach(value => {
    if (value.phoneData.origem === origin)
      mappedDestinies.push(value.phoneData.destino);
  });

  return mappedDestinies;
}

interface State {
  selectedOrigin: string | null;
  selectedDestiny: string | null;
}

const Selector: FunctionComponent = () => {
  const styles = selectorStyles();

  const phoneData: PhoneTransactionType[] = useSelector<
    GlobalState,
    PhoneTransactionType[]
  >(state => state.phoneData.transactions);

  const [state, setState] = useState<State>({
    selectedOrigin: null,
    selectedDestiny: null
  });

  function selectOrigin(
    event: React.ChangeEvent<{ name?: string; value: unknown }>
  ) {
    const selectedOrigin = event.target.value as string;
    setState(value => ({
      ...value,
      selectedOrigin: selectedOrigin.length > 0 ? selectedOrigin : null
    }));
  }

  function selectDestiny(
    event: React.ChangeEvent<{ name?: string; value: unknown }>
  ) {
    const selectedDestiny = event.target.value as string;
    setState(value => ({
      ...value,
      selectedDestiny: selectedDestiny.length > 0 ? selectedDestiny : null
    }));
  }

  return (
    <section className={styles.section}>
      <FormControl className={styles.selector1} variant="outlined">
        <InputLabel htmlFor="input-origem">Origem</InputLabel>
        <Select
          onChange={selectOrigin}
          value={state.selectedOrigin || ''}
          input={
            <OutlinedInput name="origem" id="input-origem" labelWidth={64} />
          }
        >
          <MenuItem value="">Nenhum</MenuItem>
          {(state.selectedDestiny != null
            ? mapOriginsFromDestiny(phoneData, state.selectedDestiny)
            : Array.from(new Set(phoneData.map(v => v.phoneData.origem))).sort()
          ).map((value, index) => (
            <MenuItem key={`${value + index}`} value={value}>
              {value}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      <FormControl className={styles.selector2} variant="outlined">
        <InputLabel htmlFor="input-destino">Destino</InputLabel>
        <Select
          onChange={selectDestiny}
          value={state.selectedDestiny || ''}
          input={
            <OutlinedInput name="destino" id="input-destino" labelWidth={64} />
          }
        >
          <MenuItem value="">Nenhum</MenuItem>
          {(state.selectedOrigin != null
            ? mapDestiniesFromOrigin(phoneData, state.selectedOrigin)
            : Array.from(
                new Set(phoneData.map(v => v.phoneData.destino))
              ).sort()
          ).map((value, index) => (
            <MenuItem key={`${value + index}`} value={value}>
              {value}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      <Button
        color="secondary"
        variant="contained"
        className={styles.button}
        disabled={!state.selectedOrigin || !state.selectedDestiny}
      >
        <Search />
      </Button>
    </section>
  );
};

export default Selector;
