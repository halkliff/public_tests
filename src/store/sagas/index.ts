import { takeEvery } from 'redux-saga/effects';
import { ActionTypes } from 'store/ducks/phoneData';
import phoneData from './phoneData';

export default function*() {
  yield takeEvery(ActionTypes.PRELOAD, phoneData);
}
