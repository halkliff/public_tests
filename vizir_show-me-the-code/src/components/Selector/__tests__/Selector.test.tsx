import React from 'react';
import ReactDOM from 'react-dom';
import { create, ReactTestRenderer } from 'react-test-renderer';

// This component is dependent of the redux store, therefore we must have the store
// provider available for the component.

import { Provider } from 'react-redux';
import store from 'store';

import Selector from '../Selector';

describe('Selector Component tests', () => {
  it('Renders the component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Selector />
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders the component correctly', () => {
    const selector: ReactTestRenderer = create(
      <Provider store={store}>
        <Selector />
      </Provider>
    );
    expect(selector.toJSON()).toMatchSnapshot();
  });
});
