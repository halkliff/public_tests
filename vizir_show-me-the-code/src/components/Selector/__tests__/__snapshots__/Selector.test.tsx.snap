// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Component tests Renders the component correctly 1`] = `
<section
  className="makeStyles-section-102"
>
  <div
    className="MuiFormControl-root makeStyles-selector1-103"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="input-origem"
    >
      Origem
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiSelect-outlined"
        id="select-origem"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        id="input-origem"
        name="origem"
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-181 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 36,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-182"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root makeStyles-selector2-104"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="input-destino"
    >
      Destino
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiSelect-outlined"
        id="select-destino"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        id="input-destino"
        name="destino"
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-181 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 36,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-182"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root makeStyles-button-105 MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
  </button>
</section>
`;
